test/e2e/react-compiler/next.config.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * @type {import('next').NextConfig}
+ */
+const nextConfig = {
+  experimental: {
+    // Forces nextjs to use experimental react which has the useMemoCache hook.
```

Comment:
if this is needed, wouldn't we want the `experimental.reactCompiler` flag just opt into experimental react?

Reply:
No, since we don't know what user really wants: currently `taint` or `ppr` enables experimental, and user should be able to opt in what they need. Moreover I believe this constraint will be lifted soon by compiler pkg updates.

Reply:
I don't understand. Isn't the compiler using `react/compiler-runtime` anyway, which AFAICT exists both on stable AND experimental?

this is from current canary:
```
~/code/next.js $ node -p 'require("./packages/next/dist/compiled/react/compiler-runtime")'
{ c: [Function: useMemoCache] }
~/code/next.js $ node -p 'require("./packages/next/dist/compiled/react-experimental/compiler-runtime")'
{ c: [Function: useMemoCache] }
```
So it seems that there's no need to opt into experimental here?

Reply:
This test case is originated from previous compiler example test case, os it is possible to be outdated already. I'll check.

Reply:
Ok, looks like it is outdated and not needed anymore.

Reply:
actually I was wrong: https://github.com/vercel/next.js/actions/runs/9102060902/job/25020819078?pr=65804#step:28:152

I don't know why this didn't occur on local testing 🤔

Reply:
I'll bring this back for now.